types/lib/client-options.d.ts

```diff
@@ -0,0 +1,124 @@
+import { MqttClient } from './client'
+import { Store } from './store'
+import { QoS } from './types'
+
+export interface IClientOptions extends ISecureClientOptions {
+  // The options are extended by url.parse() fields in connect() function
```

Comment:
@mcollina 

What do you want exposed here? And once we've decided, should we only extend the IClientOptions with those, rather than 
just grabbing the kitchen sink?

```
> u.parse('mqtt://www.broker.com/mqtt')
Url {
  protocol: 'mqtt:',
  slashes: true,
  auth: null,
  host: 'www.broker.com',
  port: null,
  hostname: 'www.broker.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/mqtt',
  path: '/mqtt',
  href: 'mqtt://www.broker.com/mqtt' }
```

I'll switch to TypeScript branch and `find usages` and report back